### PR TITLE
Removes JetpackSiteRef from the domain registration flows.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Removes JetpackSiteRef from the domain registration flows.

This helps simplify the code a bit and actually make it better since `JetpackSiteRef` wasn't really needed.

## To test:

1. Test claiming the free domain that comes with a paid plan from the "Register domain cell"
2. Test claiming the free domain that comes with a paid plan from the Domain Dashboard
3. Test registering a domain with a free plan from the Domain Dashboard.

## Regression Notes

1. Potential unintended areas of impact

None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift [hunk 1]
```diff
@@ -184,7 +184,7 @@ class RegisterDomainDetailsViewModel {
         let contactInformation = jsonRepresentation()
         let privacyEnabled = privacySectionSelectedItem() == CellIndex.PrivacyProtection.privately
         let registerDomainService = registerDomainDetailsService
-        let siteID = site.siteID
+        let siteID = siteID
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift:187]: Can we just remove this line?

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionViewControllerWrapper.swift [hunk 2]
```diff
@@ -25,7 +25,7 @@ final class DomainSuggestionViewControllerWrapper: UIViewControllerRepresentable
 
         let viewController = RegisterDomainSuggestionsViewController
         /// TODO: - DOMAINS - Resolve the force unwrap here
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionViewControllerWrapper.swift]: I think we can remove this comment now

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift [hunk 3]
```diff
@@ -159,15 +159,12 @@ class RegisterDomainSuggestionsViewController: UIViewController {
...
+            vc.freeSiteAddress = site.freeSiteAddress
 
-            if let blog = BlogService.blog(with: site) {
-                vc.domainType = domainType
-                vc.freeSiteAddress = blog.freeSiteAddress
-
-                if blog.hasBloggerPlan == true {
-                    vc.domainSuggestionType = .allowlistedTopLevelDomains(["blog"])
-                }
+            if site.hasBloggerPlan == true {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift]: Nitpick, but we could take the `== true` out of here

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift [hunk 4]
```diff
@@ -293,11 +300,12 @@ extension RegisterDomainSuggestionsViewController: NUXButtonViewControllerDelega
     }
 
     private func presentWebViewForCurrentSite(domainSuggestion: DomainSuggestion) {
-        guard let siteUrl = URL(string: "\(site.homeURL)"), let host = siteUrl.host,
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift:296]: We could also do `URL(string: homeURL as String)`

**Review by frosty — COMMENTED** [review]
Functionally this is working as expected, and looks much cleaner 👍 Just a few small code suggestions.

**Comment by mokagio** [comment]
I'm going to bump this to the next release because we'll be code freezing 18.5 today and it seems like this needs a bit more touches.

From the looks of it, it seems like this might be something we want in 18.5, though? If this cannot wait two weeks and it's important that it makes it into this release, let me know and we'll organize a new beta once ready. 👍 😃

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift (hunk 1 above)
Comment (diegoreymendez) [reply WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift:187]: <strike>I'd rather keep it to only retain the site ID within the closure below and avoid retaining self.</strike>

What I said above is kind of correct but let me rephrase: by using this approach I am able to avoid having to unwrap self in the closures below when I need to pass `siteID` as parameter to the purchase service call.

**Comment by diegoreymendez** [comment]
@frosty - Feedback addressed.

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift:187]: Ah, I see! 👍

**Review by frosty — APPROVED** [review]
Changes look good!

**Comment by diegoreymendez** [comment]
Thank you!
